presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1027,6 +1038,139 @@ core::LocalPartitionNode::Type toLocalExchangeType(
       VELOX_UNSUPPORTED("Unsupported exchange type: {}", toJsonString(type));
   }
 }
+
+std::vector<std::shared_ptr<const connector::hive::HiveColumnHandle>>
+toHiveColumns(
+    const protocol::List<protocol::HiveColumnHandle>& inputColumns,
+    bool& hasPartitionColumn) {
+  hasPartitionColumn = false;
+  std::vector<std::shared_ptr<const connector::hive::HiveColumnHandle>>
+      hiveColumns;
+  hiveColumns.reserve(inputColumns.size());
+  for (const auto& columnHandle : inputColumns) {
+    hasPartitionColumn |=
+        columnHandle.columnType == protocol::ColumnType::PARTITION_KEY;
+    hiveColumns.emplace_back(
+        std::dynamic_pointer_cast<connector::hive::HiveColumnHandle>(
+            toColumnHandle(&columnHandle)));
+  }
+  return hiveColumns;
+}
+
+HiveBucketProperty::Kind toHiveBucketPropertyKind(
+    protocol::BucketFunctionType bucketFuncType) {
+  switch (bucketFuncType) {
+    case protocol::BucketFunctionType::PRESTO_NATIVE:
+      return HiveBucketProperty::Kind::kPrestoNative;
+    case protocol::BucketFunctionType::HIVE_COMPATIBLE:
+      return HiveBucketProperty::Kind::kHiveCompatible;
+    default:
+      VELOX_USER_FAIL(
+          "Unknown hive bucket function: {}", toJsonString(bucketFuncType));
+  }
+}
+
+std::vector<std::string> fromStringList(
+    const protocol::List<protocol::String>& stringList) {
```

Comment:
Is this needed List<String> is same as std::vector<std::string>.

```
template <typename T>
using List = std::vector<T>;

using String = std::string;
```

Reply:
Ah, I thought it is a different data structure. Thanks!